internal/provider/job_action.go

```diff
@@ -0,0 +1,169 @@
+package provider
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/ansible/terraform-provider-aap/internal/provider/customtypes"
+	"github.com/hashicorp/terraform-plugin-framework/action"
+	"github.com/hashicorp/terraform-plugin-framework/action/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+)
+
+// JobAction represents a job action that can be executed in AAP.
+type JobAction struct {
+	client ProviderHTTPClient
+}
+
+func NewJobAction() action.Action {
+	return &JobAction{}
+}
+
+var (
+	_ action.Action = (*JobAction)(nil)
+)
+
+type JobActionModel struct {
+	JobModel
+	IgnoreJobResults types.Bool `tfsdk:"ignore_job_results"`
+}
+
+// Schema defines the schema for the job action
+func (a *JobAction) Schema(_ context.Context, _ action.SchemaRequest, resp *action.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"job_template_id": schema.Int64Attribute{
+				Required:    true,
+				Description: "ID of the job template.",
+			},
+			"inventory_id": schema.Int64Attribute{
+				Optional: true,
+				Description: "Identifier for the inventory where job should be created in. " +
+					"If not provided, the job will be created in the default inventory.",
+			},
+			"extra_vars": schema.StringAttribute{
+				Description: "Extra Variables. Must be provided as either a JSON or YAML string.",
+				Optional:    true,
+				CustomType:  customtypes.AAPCustomStringType{},
+			},
+			"wait_for_completion": schema.BoolAttribute{
+				Optional: true,
+				Description: "When this is set to `true`, Terraform will wait until this aap_job resource is created, reaches " +
```

Comment:
If you're writing Markdown in a description, it's more accurate to use `MarkdownDescription` instead of `Description`. It looks like `job.md` is still rendering as expected, so this isn't causing a problem, just FYI

See https://developer.hashicorp.com/terraform/plugin/framework/handling-data/schemas#markdowndescription